File: core/src/main/java/com/lomekwi/cave/pipeline/Sink.java
package com.lomekwi.cave.pipeline;

import org.jspecify.annotations.*;

public final class Sink extends Node{
    private final InPort<Object> in = addInPort(new InPort<>("输入"));
    @Override
    public @NonNull String getName() {
        return "总输出";
    }
    public @Nullable Object get(){
        return in.getData();
    }
}

File: core/src/main/scala/com/lomekwi/cave/pipeline/Sink.scala
package com.lomekwi.cave.pipeline

final class Sink extends Node {
  private final val in: Node.InPort[Object] = addInPort(new Node.InPort[Object]("输入"))

  override def getName: String = {
    "总输出"
  }

  def getIn: Node.InPort[Object] = in

  def get(): Object = {
    in.getData
  }

  override def canRemove: Boolean = false
}
